### PR TITLE
Temple / call creation with firestore doc subscription 

### PR DESCRIPTION
**Pull request by Nipher**
- [x] Endpoint to create a DailyCo room and a live-conent-session to be used among participants (with firestore onSnapshot)
- [x] List of rooms
  - [x] Fetch and render rooms
  - [x] Loader/refresher
- [x] Create room
  - [x] Create room and re-fetch rooms list
  - [x] Allow custom name (somehow doesn't work to do it like dailyco api docs says)
  - [x] Create live content doc (firebase complaining about app not initialized)
- [x] Join room
  - [x]  Live content doc snapshot sync
    - [x] Emulator
    - [x] Staging
  - [x] Join room call on multiple devices (test)

### CODE REVIEW

**Comment by Nipher** [comment]
Seems like when I run it locally the client firestore is not connecting to the emulator, can't find documents, etc...
not sure why, it worked ok a while ago

**Comment by Nipher** [comment]
Tested running onSnapshot API against staging and it works fine.

Not sure what's up, it seems like the app is not connecting to the Firestore emulator, the firebase rules seem correct, the auth works through the emulator, but firestore somehow "fails" silently, almost like it is connecting to some other firestore with no data in it. 🤔

File: client/__mocks__/@react-native-firebase/firestore.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const mockFirestore = {
```
Comment (Nipher) [inline client/__mocks__/@react-native-firebase/firestore.js:1]: Maybe possible and better to use the real emulator with "seeded mock data" instead 🤓

File: client/src/routes/Temples/Temples.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  padding: SPACINGS.SIXTEEN,
+  borderRadius: SPACINGS.TWELVE,
+  backgroundColor: COLORS.WHITE_TRANSPARENT,
+});
+
+type ScreenNavigationProps = NativeStackNavigationProp<ScreenProps>;
+
+const Temples = () => {
+  const {t} = useTranslation(NS.SCREEN.TEMPLES);
+  const {fetchTemples, isLoading, addTemple, temples} = useTemples();
```
Comment (Nipher) [inline client/src/routes/Temples/Temples.tsx:40]: thoughts?

File: client/android/app/build.gradle [hunk 3]
```diff
@@ -282,6 +282,12 @@ android {
 }
 
 dependencies {
+    // Import the BoM for the Firebase platform
+    implementation platform('com.google.firebase:firebase-bom:30.3.0')
+
+    implementation 'com.google.firebase:firebase-auth'
+    implementation 'com.google.firebase:firebase-firestore'
```
Comment (gewfy) [inline client/android/app/build.gradle:289]: I'm pretty sure that these lines are not needed as React Native Firebase is auto linked including the Firebase native lib. Do you reckon why this was added?

Comment (Nipher) [reply client/android/app/build.gradle:289]: Probably from one of firebase docs I checked while trying to understand how things work and make it work on my machine

But it was probably a native code documentation, React Native Firebase doesn't mentions it, could probably be removed 👍

**Review by gewfy — APPROVED** [review]
e578293034d9d0699efa1bc4c269088fc68b68a9_00 (1)

**Review by gewfy — APPROVED** [review]
MYUM

**Review by gewfy — APPROVED** [review]
4Bi1

**Pull request merged**